x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CcrAutoFollowInfoFetcher.java

```diff
@@ -0,0 +1,140 @@
+                List<AutoFollowedIndex> autoFollowedIndices =
+                    remoteClusterAutoFollowedIndices.computeIfAbsent(remoteCluster, unused -> new ArrayList<>());
+
+                autoFollowedIndices.add(new AutoFollowedIndex(leaderIndexName, followerIndexName));
+            }
+        }
+
+        if (remoteClusterAutoFollowedIndices.isEmpty()) {
+            listener.onResponse(Collections.emptyList());
+            return;
+        }
+
+        GroupedActionListener<List<String>> clusterResponsesListener = new GroupedActionListener<>(
+            listener.map((clusterAutoFollowedSystemIndices) ->
+                clusterAutoFollowedSystemIndices.stream()
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toList())),
+            remoteClusterAutoFollowedIndices.size()
+        );
+
+        // When a system index is followed we don't copy over the isSystem flag into IndexMetadata
+        // We need to fetch the remote cluster state in order to check whether or not the following index
+        // follows a leader system index.
+        for (Map.Entry<String, List<AutoFollowedIndex>> remoteAutoFollowedIndicesEntry : remoteClusterAutoFollowedIndices.entrySet()) {
+            final String remoteCluster = remoteAutoFollowedIndicesEntry.getKey();
+            final List<AutoFollowedIndex> autoFollowedIndices = remoteAutoFollowedIndicesEntry.getValue();
+
+            try {
+                client.getRemoteClusterClient(remoteCluster)
+                    .admin()
+                    .cluster()
+                    .prepareState()
+                    .clear()
+                    .setMetadata(true)
+                    .execute(new ActionListener<ClusterStateResponse>() {
+                        @Override
+                        public void onResponse(ClusterStateResponse stateResponse) {
+                            final ClusterState clusterState = stateResponse.getState();
+                            List<String> autoFollowedSystemIndices = new ArrayList<>();
+                            for (AutoFollowedIndex autoFollowedIndex : autoFollowedIndices) {
+                                final IndexAbstraction indexAbstraction =
+                                    clusterState.metadata().getIndicesLookup().get(autoFollowedIndex.remoteIndexName);
+                                if (indexAbstraction != null && indexAbstraction.isSystem()) {
+                                    autoFollowedSystemIndices.add(autoFollowedIndex.localIndexName);
+                                }
+                            }
+                            clusterResponsesListener.onResponse(autoFollowedSystemIndices);
+                        }
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            clusterResponsesListener.onFailure(e);
```

Comment:
I think this bails out of the deprecation check and I wonder if a remote cluster might not be somewhat expected to be unavailable sometimes. I wonder if we should instead present something about remote cluster not available in deprecation info?

Reply:
The deprecation API doesn't provide a way to return fine grained error information. Currently the only thing we can do is return a `DeprecationIssue` but I would find a bit confusing to present an error about a remote cluster unavailability as a `DeprecationIssue`.